app/scripts/controllers/preferences-controller.ts

```diff
@@ -149,125 +159,289 @@ export type PreferencesControllerState = {
-   * @property initState.currentLocale The preferred language locale key
-   * @property initState.selectedAddress A hex string that matches the currently selected address in the app
-   */
-  constructor(opts: PreferencesControllerOptions) {
+export default class PreferencesController extends BaseController<
+  typeof controllerName,
+  PreferencesControllerState,
+  PreferencesControllerMessenger
+> {
+  /**
+   * Constructs a Preferences controller.
+   *
+   * @param options - the controller options
+   * @param options.networkConfigurationsByChainId - The network configurations
+   * @param options.initLangCode - The language code
+   * @param options.messenger - The controller messenger
+   * @param options.state - The initial controller state
+   */
+  constructor({
+    networkConfigurationsByChainId,
+    initLangCode,
+    messenger,
+    state,
+  }: PreferencesControllerOptions) {
     const addedNonMainNetwork: Record<Hex, boolean> = Object.values(
-      opts.networkConfigurationsByChainId ?? {},
+      networkConfigurationsByChainId ?? {},
     ).reduce((acc: Record<Hex, boolean>, element) => {
       acc[element.chainId] = true;
       return acc;
     }, {});
-
-    const initState: PreferencesControllerState = {
-      selectedAddress: '',
-      useBlockie: false,
-      useNonceField: false,
-      usePhishDetect: true,
-      dismissSeedBackUpReminder: false,
-      useMultiAccountBalanceChecker: true,
-      useSafeChainsListValidation: true,
-      // set to true means the dynamic list from the API is being used
-      // set to false will be using the static list from contract-metadata
-      useTokenDetection: opts?.initState?.useTokenDetection ?? true,
-      useNftDetection: opts?.initState?.useTokenDetection ?? true,
-      use4ByteResolution: true,
-      useCurrencyRateCheck: true,
-      useRequestQueue: true,
-      openSeaEnabled: true,
-      securityAlertsEnabled: true,
-      watchEthereumAccountEnabled: false,
-      bitcoinSupportEnabled: false,
-      bitcoinTestnetSupportEnabled: false,
-      ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
-      addSnapAccountEnabled: false,
-      ///: END:ONLY_INCLUDE_IF
-      advancedGasFee: {},
-
-      // WARNING: Do not use feature flags for security-sensitive things.
-      // Feature flag toggling is available in the global namespace
-      // for convenient testing of pre-release features, and should never
-      // perform sensitive operations.
-      featureFlags: {},
-      incomingTransactionsPreferences: {
-        ...mainNetworks,
-        ...addedNonMainNetwork,
-        ...testNetworks,
+    super({
+      messenger,
+      metadata: controllerMetadata,
+      name: controllerName,
+      state: {
+        ...getDefaultPreferencesControllerState(),
+        incomingTransactionsPreferences: {
+          ...mainNetworks,
+          ...addedNonMainNetwork,
+          ...testNetworks,
+        },
+        currentLocale: initLangCode ?? '',
```

Comment:
Shouldn't this be part of the `state` passed from MetamaskController? I see that `currentLocale` is also part of the persisted state

Reply:
This to say: do you think it would make sense to simply remoive `initLangCode` from the constructor options?